packages/neo-one-monitor/src/Reporter.js

```diff
@@ -0,0 +1,153 @@
+      name: 'browser_request_counter',
+      help: 'Counts fetch requests from Browser Reporter to endpoint',
+      labelNames: [KNOWN_LABELS.ERROR],
+    });
+    const emptyBackReport = {
+      logs: [],
+      metrics: {
+        counters: {},
+        histograms: {},
+      },
+    };
+
+    this._subscription = interval(timer)
+      .pipe(
+        mergeScan(
+          async backReport => {
+            const logs = logger.collect();
+            const metrics = metricsFactory.collect();
+
+            const report = this._addBackReport({ logs, metrics }, backReport);
+
+            const response = await this._report(report);
+
+            if (!response.ok) {
+              requestCounter.inc({ [KNOWN_LABELS.ERROR]: true });
+              return this._constructBackReport(report);
+            }
+            requestCounter.inc({ [KNOWN_LABELS.ERROR]: false });
+            return emptyBackReport;
+          },
+          emptyBackReport,
+          1,
+        ),
+      )
+      .subscribe();
+  }
+
+  async _report(report: Report): Promise<Response> {
+    return fetch(this._endpoint, {
+      method: 'POST',
+      body: JSON.stringify(report),
+    });
+  }
+
+  _addBackReport(report: Report, backReport?: Report): Report {
+    if (backReport == null) {
+      return report;
+    }
+    const logs = backReport.logs.concat(report.logs);
+    const { metrics } = report;
+
+    const newMetrics = utils
+      .keys(backReport.metrics)
+      .reduce((accMetricType, metricType) => {
+        const resultMetricType = { ...accMetricType };
+        resultMetricType[metricType] = utils
+          .keys(backReport.metrics[metricType])
+          .reduce(
+            (accMetric, name) => {
+              const resultMetric = { ...accMetric };
+              if (metrics[metricType][name] != null) {
+                const resultValues = backReport.metrics[metricType][
+                  name
+                ].values.concat(metrics[metricType][name].values);
+                resultMetric[name].values = resultValues;
+              } else {
+                resultMetric[name] = backReport.metrics[metricType][name];
+              }
+              return resultMetric;
+            },
+            { ...metrics[metricType] },
+          );
+        return resultMetricType;
+      }, {});
+
+    return {
+      logs,
+      metrics: {
+        counters: newMetrics.counters,
+        histograms: newMetrics.histograms,
+      },
+    };
+  }
+
+  _constructBackReport(report: Report): Report {
+    const backLogs = report.logs.slice(-MAX_BACKLOG);
+
+    const backMetrics = utils
+      .keys(report.metrics)
+      .reduce((accMetricType, metricType) => {
+        const resultMetricType = { ...accMetricType };
+        resultMetricType[metricType] = utils
+          .keys(report.metrics[metricType])
+          .reduce((accMetric, name) => {
+            const result = { ...accMetric };
+            result[name] = report.metrics[metricType][name];
```

Comment:
We need to slice something here right? Otherwise this is just cloning the input metrics. Probably a case where you should be using `forEach` since you need to modify the metric values but not actually change the overall structure of `report.metrics`.